Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: 
Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. 
Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. 
Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.
Прикрепить ссылку на курсовой проект на github. (см. выбор темы в ДЗ к Уроку 6)

In [150]:
import numpy as np
from scipy import stats
FP=np.array([173,175,180,178,177,185,183,182])
HP=np.array([177,179,180,188,177,172,171,184,180])
WP=np.array([172,173,169,177,166,180,178,177,172,166,170])

In [59]:
y1=FP.mean()
y1

179.125

In [60]:
y2=HP.mean()
y2

178.66666666666666

In [61]:
y3=WP.mean()
y3

172.72727272727272

In [62]:
Y=np.sum(np.sum(FP)+np.sum(HP)+np.sum(WP))/(len(FP)+len(HP)+len(WP))
Y

176.46428571428572

In [63]:
Sf=((y1-Y)**2)*len(FP)+((y2-Y)**2)*len(HP)+((y3-Y)**2)*len(WP)
Sf

253.9074675324678

In [117]:
ans=0
for a in range(len(FP)):
    ans+=np.sum((FP[a]-y1)**2)
    print (FP[a],'-',y1,'=',FP[a]-y1,'E= ',ans) 

173 - 179.125 = -6.125 E=  37.515625
175 - 179.125 = -4.125 E=  54.53125
180 - 179.125 = 0.875 E=  55.296875
178 - 179.125 = -1.125 E=  56.5625
177 - 179.125 = -2.125 E=  61.078125
185 - 179.125 = 5.875 E=  95.59375
183 - 179.125 = 3.875 E=  110.609375
182 - 179.125 = 2.875 E=  118.875


In [130]:
def S_ost_one_only (input_array,mean_in_array):
    answ=0
    for a in range(len(input_array)):
        answ+=np.sum((input_array[a]-mean_in_array)**2)
    return answ        

In [136]:
Sost=S_ost_one_only(FP,y1)+S_ost_one_only(HP,y2)+S_ost_one_only(WP,y3)
Sost

577.0568181818181

In [137]:
Sf+Sost

830.9642857142859

In [147]:
Sf/2

126.9537337662339

In [148]:
Sost/(28-3)

23.082272727272724

In [149]:
(Sf/2)/(Sost/(28-3))

5.500053450812599

df2=k-1=3-1=2—число степеней свободы меньшей дисперсии
df1=n-k=28-3=25 число степеней свободы большей дисперсии
k=2, n=25
Для таблицы Фишера при alpha=0.05 критическое значение = 3.3852
Так как 5.500053450812599 > 3.3852 следовательно различе между группами статистически значемо.

In [ ]:
Для проверки.

In [152]:
stats.f_oneway(FP,HP,WP)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)